### PR TITLE
Prime1 elevators

### PR DESCRIPTION
**Pull request by FractureFalcon**
Added custom names to the Prime 1 elevators to greatly improve the accessibility of the elevators tab on the tracker. Included a few UI tweaks to help with the display.

### CODE REVIEW

**Comment by henriquegemignani** [comment]
Instead of adding Prime 1 related things to the Echoes focused `echoes_teleporters.py`, the tracker window should be changed to also check `prime1_elevators.py`.

That `setup_elevators` function is assuming the game is Echoes and doing things that are Echoes-specific.

**Comment by FractureFalcon** [comment]
> Instead of adding Prime 1 related things to the Echoes focused `echoes_teleporters.py`, the tracker window should be changed to also check `prime1_elevators.py`.
> 
> That `setup_elevators` function is assuming the game is Echoes and doing things that are Echoes-specific.

i'm unfamiliar with the needs of games other than prime and echoes; does this logic need to support prime 3 / super? are you fine with the tracker_window logic just being an if check on the current game (if prime1 else echoes)? i agree with your point here, was just unsure how generic the elevator tracking logic needed to be.

**Comment by henriquegemignani** [comment]
It works very well if you could make it so that if it's a game with custom elevator names, these are used. Otherwise it falls back to the default names.

**Comment by FractureFalcon** [comment]
in that case, what would you think of splitting the elevator name overrides into a prime and echoes array, and having the logic for using the right one in `elevator_area_name`? then just renaming that class to be something like `elevators.py`? it strikes me as fairly generic as it is

**Comment by henriquegemignani** [comment]
That module's more of a static data for each game. To be honest, these elevator names are a mess right now. `randovania/gui/preset_settings/elevators_tab.py:_create_check_for_source_elevator` does it differently as well. Might just go with that style as well.

**Comment by duncathan** [comment]
be sure to add an entry to changelog.md as well

**Comment by FractureFalcon** [comment]
hey, i'm pretty much done with this. but i got stuck with one weird thing. if i change what the elevators are called, even one character, the prime 1 patcher breaks. i can maintain the existing names as a work around, but... i feel uncomfortable leaving it be. do you know why it's like that?

**Comment by henriquegemignani** [comment]
The prime 1 elevator names needs to remain that way yes, it's part of the randomprime api. We can remove the `\x00` before sending to Qt, as that breaks though.

**Comment by henriquegemignani** [comment]
As duncathan said, could you also add an entry to the changelog please?

**Comment by FractureFalcon** [comment]
lol i had, i think i forgot to push it. should be there now

**Comment by FractureFalcon** [comment]
i'll look at what's up with the tests after work

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,10 @@ time, before giving the error message of why it failed. It's useful for investig
 - Changed: Preset summaries have been split better into categories.
 - Removed: The "Never" option for dangerous actions has been removed from the UI, as it currently doesn't work.
 
+### Custom Elevator Names (Prime 1 + Echoes)
+- Changed: Reworked how elevator custom names are handled in the UI
+- Added: Support for short names in the tracker
```
Comment (henriquegemignani) [inline CHANGELOG.md:36]: The Changelog is written from the perspective of an end-user. So pure code changes are ignored.
This change would be something like "Added: Prime 1 Elevators in the tracker now have nicknames" or something similar.

Also, this doesn't need a full section to it, should just show up in the general section of the next `## [Unreleased]` block.

**Comment by FractureFalcon** [comment]
actually, re: the test failures, they seem to be originating in `area_by_area_location`; which, if you look at prime1_elevators.py, i have a section in there labeled "Here to prevent exceptions in _get_elevator_area_name"; the exception they're preventing is the key exception when that function tries to pass the area into the world list to fetch the name when the custom name isn't there, which is the same failure popping up in the tests.

but, while i was debugging it, i dug through it until i _found_ the Artifact Temple (one of the areas it can't find in the world list), and i have no idea why it doesn't match it correctly

do you know what's up with that? did i somehow mess with the world list in my changeset?

**Comment by henriquegemignani** [comment]
Have you ran the tests locally on your computer?

Checking from the output in the CI event, I can see this test failure:

```
_________ test_on_preset_changed[RandovaniaGame.METROID_PRIME_ECHOES] __________

skip_qtbot = <pytestqt.qtbot.QtBot object at 0x7f3e289e1ac0>
preset_manager = <randovania.interface_common.preset_manager.PresetManager object at 0x7f3e289e1790>
game = <RandovaniaGame.METROID_PRIME_ECHOES: 'prime2'>

    @pytest.mark.parametrize("game", [RandovaniaGame.METROID_PRIME, RandovaniaGame.METROID_PRIME_ECHOES, RandovaniaGame.METROID_PRIME_CORRUPTION])
    def test_on_preset_changed(skip_qtbot, preset_manager, game):
        # Setup
        base = preset_manager.default_preset_for_game(game).get_preset()
        preset = dataclasses.replace(base,
                                     uuid=uuid.UUID('b41fde84-1f57-4b79-8cd6-3e5a78077fa6'),
                                     base_preset_uuid=base.uuid)
        editor = PresetEditor(preset)
>       window = LogicSettingsWindow(None, editor)

test/gui/dialog/test_logic_settings_window.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
randovania/gui/preset_settings/logic_settings_window.py:38: in __init__
    self._tabs = game_specific_gui.preset_editor_tabs_for(editor, window_manager)
randovania/gui/game_specific_gui.py:134: in preset_editor_tabs_for
    return TAB_PROVIDER_FOR_GAME[editor.game](editor, window_manager)
randovania/gui/game_specific_gui.py:74: in prime2_preset_tabs
    PresetElevators(editor, game_description),
randovania/gui/preset_settings/elevators_tab.py:46: in __init__
    self._create_source_elevators()
randovania/gui/preset_settings/elevators_tab.py:117: in _create_source_elevators
    checks: Dict[Teleporter, QtWidgets.QCheckBox] = {
randovania/gui/preset_settings/elevators_tab.py:118: in <dictcomp>
    loc: self._create_check_for_source_elevator(loc) for loc in locations
randovania/gui/preset_settings/elevators_tab.py:92: in _create_check_for_source_elevator
    name = elevators.get_elevator_area_name(self.game_enum, self.game_description.world_list, area, False)
randovania/games/prime/elevators.py:22: in get_elevator_area_name
    return _get_elevator_area_name(CUSTOM_NAMES_FOR_ELEVATORS, game, world_list, area_location, include_world_name)
randovania/games/prime/elevators.py:45: in _get_elevator_area_name
    area = world_list.area_by_area_location(area_location)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <randovania.game_description.world.world_list.WorldList object at 0x7f3e34072b50>
location = Area[Aerie]

    def area_by_area_location(self, location: AreaLocation) -> Area:
>       return self._ids_to_area[location]
E       KeyError: Area[Aerie]

randovania/game_description/world/world_list.py:265: KeyError
```

`location` is an Area, not an AreaLocation.

**Comment by FractureFalcon** [comment]
> `location` is an Area, not an AreaLocation.

thanks i would have gone the rest of my life not noticing that

**Pull request merged**